src/lib.rs

```diff
@@ -667,6 +667,32 @@ impl RawInvoice {
 			num_traits => None,
 		}).collect::<Vec<&Route>>()
 	}
+
+	pub fn amount_pico_btc(&self) -> Option<u64> {
+		match self.hrp.raw_amount {
```

Comment:
Imo using functional combinators like `map` or `flat_map` (or however they are called in rust) these nested matches could be avoided.